MQ2FeedMe.cpp

```diff
@@ -342,23 +684,131 @@ PLUGIN_API void SetGameState(DWORD GameState)
 			WritePrivateProfileString("Settings", "Announce", bAnnLevels ? "1" : "0", INIFileName);
 			WritePrivateProfileString("Settings", "FoodWarn", bFoodWarn ? "1" : "0", INIFileName);
 			WritePrivateProfileString("Settings", "DrinkWarn", bDrinkWarn ? "1" : "0", INIFileName);
-			if (!Loaded) {
-				ReadList(&Hunger,"FOOD");
-				ReadList(&Thirst,"DRINK");
+			WritePrivateProfileString("Settings", "IgnoreSafeZones", bIgnoreSafeZones ? "1" : "0", INIFileName);
+
+			if (!Loaded)
+			{
+				ReadList(&Hunger, "FOOD");
+				ReadList(&Thirst, "DRINK");
 				Loaded = true;
 			}
 		}
 	}
 }
 
+void FeedMeImGuiSettingsPanel()
+{
+	if (ImGui::Checkbox("Announce", &bAnnConsume))
+	{
+		WritePrivateProfileBool("Settings", "Announce", bAnnConsume, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Announce Levels and Consumption.");
+
+	if (ImGui::Checkbox("IgnoreSafeZones", &bIgnoreSafeZones))
+	{
+		WritePrivateProfileBool("Settings", "IgnoreSafeZones", bIgnoreSafeZones, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Ignore Safe Zones like poknowledge or guild lobby for auto consumption.");
+
+	if (ImGui::Checkbox("Food Warn", &bFoodWarn))
+	{
+		WritePrivateProfileBool("Settings", "FoodWarn", bFoodWarn, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Warn when no food.");
+
+	if (ImGui::Checkbox("Drink Warn", &bFoodWarn))
+	{
+		WritePrivateProfileBool("Settings", "DrinkWarn", bFoodWarn, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Warn when no drink.");
+
+	if (ImGui::InputInt("DrinkAt", &iDrinkAt)) {
+		iDrinkAt = std::clamp(iDrinkAt, 0, 5000);
+
+		WritePrivateProfileInt("Settings", "AutoDrink", iDrinkAt, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Thirst level to drink.");
+
+	if (ImGui::InputInt("FeedAt", &iFeedAt)) {
+		iFeedAt = std::clamp(iFeedAt, 0, 5000);
+
+		WritePrivateProfileInt("Settings", "AutoFeed", iFeedAt, INIFileName);
+	}
+	ImGui::SameLine();
+	mq::imgui::HelpMarker("Hunger level to eat.");
+
+	const float buttonsize = ImGui::GetWindowSize().x * 0.45f;
+	if (ImGui::Button("Add Food From Cursor", ImVec2(buttonsize, 0)))
+	{
+		HandleFoodDrinkItem();
+	}
+	ImGui::SameLine();
+
+	if (ImGui::Button("Add Drink On Cursor", ImVec2(buttonsize, 0)))
+	{
+		HandleFoodDrinkItem();
+	}
+
+	ImGui::BeginChild("FoodChild", ImVec2(0, 150), true);
+	if (ImGui::BeginTable("##FeedMeFoodList", 1, ImGuiTableFlags_BordersInnerV))
+	{
+		ImGui::TableSetupColumn("Food");
+		ImGui::TableSetupScrollFreeze(0, 1);
+		ImGui::TableHeadersRow();
+		size_t index = 0;
+		for (const auto& food : Hunger)
+		{
+			ImGui::PushID(static_cast<int>(index));
+			ImGui::TableNextRow();
+			ImGui::TableNextColumn();
+			ImGui::Text(food.c_str());
```

Comment:
`ImGui::Text("%s", food.c_str());` or use ImGui::TextUnformatted

Text expects a format string. If food had a `%` character in it, it could be bad.